python/cudf/cudf/core/frame.py

```diff
@@ -2688,7 +2688,10 @@ def acos(self):
                     1.5707963267948966,  1.266103672779499],
                     dtype='float64')
         """
-        return self._unaryop("acos")
+        data = self.astype("float64")
+        data = data._unaryop("acos")
+        newdata = data.mask(data < 0)
+        return newdata
```

Comment:
Looks like this fixes a bug? If so, lets add a test case so we don't accidentally regress.